ride_share.rb

```diff
@@ -0,0 +1,73 @@
+# Create the data structure by starting with the outline. Then, assign this data structure to a variable. 
+# Give the variable a good name that represents what the outermost layer is
+@drivers_data = {
+    DR001: [
+    {trip_date: "2016-02-03", trip_cost: 10, rider_id: "RD0003", trip_rating: 3},
+    {trip_date: "2016-02-03", trip_cost: 30, rider_id: "RD0015", trip_rating: 4},
+    {trip_date: "2016-05-16", trip_cost: 45, rider_id: "RD0003", trip_rating: 2}
+    ],
+    DR002: [
+    {trip_date: "2016-02-03", trip_cost: 25, rider_id: "RD0073", trip_rating: 5},
+    {trip_date: "2016-02-04", trip_cost: 15, rider_id: "RD0013", trip_rating: 1},
+    {trip_date: "2016-02-05", trip_cost: 35, rider_id: "RD0066", trip_rating: 3}
+    ],
+    DR003: [
+    {trip_date: "2016-02-04", trip_cost: 5, rider_id: "RD0066", trip_rating: 5},
+    {trip_date: "2016-02-05", trip_cost: 50, rider_id: "RD0003", trip_rating: 2}
+    ],
+    DR004: [
+    {trip_date: "2016-02-03", trip_cost: 5, rider_id: "RD0022", trip_rating: 5},
+    {trip_date: "2016-02-04", trip_cost: 10, rider_id: "RD0022", trip_rating: 4},
+    {trip_date: "2016-02-05", trip_cost: 20, rider_id: "RD0073", trip_rating: 5}
+    ]
+}
+
+puts "The number of rides each driver has given is:" 
+@drivers_data.map do |driver, trips| 
+    puts "Driver #{driver}: #{trips.length} rides" 
+end 
+
+# Totals method for total rides and rating
+def total_of(key)
+    total = {}
+    @drivers_data.each do |driver, value|
+    total_2 = []
```

Comment:
On line 33, you're accessing an instance variable (though I'm guessing you didn't realize this at the time). Since we're not inside a class, `@drivers_data` is effectively acting as a global variable here. In general, this is frowned upon in Ruby. Instead, you should pass the data in as an extra parameter to this method.